enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/AuthFilter.java

```diff
@@ -366,4 +371,14 @@ private void setInterceptorAuthContextMetadata(Authenticator authenticator, Requ
         requestContext.addMetadataToMap(InterceptorConstants.AuthContextFields.KEY_TYPE,
                 Objects.toString(authContext.getKeyType(), ""));
     }
+
+    private void setAccessLogMetadata(RequestContext requestContext) {
```

Comment:
We cannot consider the apiKey to be a constant called api-key. Hence we should use the same method used in BuildResponse method. And we could populate this dynamic metadata within buildResponse method too.